### PR TITLE
[ie/ilpost] Add extractor

### PR DESCRIPTION
**Pull request by CapacitorSet**
**IMPORTANT**: PRs without the template will be CLOSED

### Description of your *pull request* and other information

Adds an extractor for podcasts from Italian online newspaper https://www.ilpost.it/.

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check all of the following options that apply:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Fix or improvement to an extractor (Make sure to add/update tests)
- [x] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

</details>

### CODE REVIEW

File: yt_dlp/extractor/ilpost.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        podcast_display_id = self._match_id(url)
+        webpage = self._download_webpage(url, podcast_display_id)
+
+        endpoint_metadata = self._search_json(start_pattern='var ilpostpodcast = ', end_pattern=';', string=webpage, name='endpoint-metadata', video_id=podcast_display_id)
+        endpoint_url = endpoint_metadata["ajax_url"]
+        ajax_cookie = endpoint_metadata["cookie"]
+        episode_id = endpoint_metadata["post_id"]
+        podcast_id = endpoint_metadata["podcast_id"]
+        # We don't need escaping because all values are POST-safe
+        post_data = f"action=checkpodcast&cookie={ajax_cookie}&post_id={episode_id}&podcast_id={podcast_id}"
```
Comment (seproDev) [inline yt_dlp/extractor/ilpost.py:49]: Use `urlencode_postdata` from `..utils`

File: yt_dlp/extractor/ilpost.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+            # * Any Python type, e.g.
+            #     'view_count': int,
+        }
+    }]
+
+    def _real_extract(self, url):
+        podcast_display_id = self._match_id(url)
+        webpage = self._download_webpage(url, podcast_display_id)
+
+        endpoint_metadata = self._search_json(start_pattern='var ilpostpodcast = ', end_pattern=';', string=webpage, name='endpoint-metadata', video_id=podcast_display_id)
```
Comment (seproDev) [inline yt_dlp/extractor/ilpost.py:43]: Line is too long. Please follow the style guidelines.

**Review by seproDev — CHANGES_REQUESTED** [review]
You also need to add a import to `_extractors.py`

File: yt_dlp/extractor/ilpost.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+        podcast_display_id = self._match_id(url)
+        webpage = self._download_webpage(url, podcast_display_id)
+
+        endpoint_metadata = self._search_json(
+            start_pattern='var ilpostpodcast = ',
+            end_pattern=';',
+            string=webpage,
+            name='endpoint-metadata',
+            video_id=podcast_display_id,
+        )
```
Comment (seproDev) [inline yt_dlp/extractor/ilpost.py:38]:
```suggestion
        endpoint_metadata = self._search_json(
            r'var\s+ilpostpodcast\s*=', webpage, 'metadata', podcast_display_id)
```

File: yt_dlp/extractor/ilpost.py [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+        episode_id = endpoint_metadata['post_id']
+        podcast_id = endpoint_metadata['podcast_id']
+        podcast_metadata = self._download_json(endpoint_metadata['ajax_url'], podcast_display_id, data=urlencode_postdata({
+            'action': 'checkpodcast',
+            'cookie': endpoint_metadata['cookie'],
+            'post_id': episode_id,
+            'podcast_id': podcast_id,
+        }))
```
Comment (seproDev) [inline yt_dlp/extractor/ilpost.py:46]:
```suggestion
        podcast_metadata = self._download_json(
            endpoint_metadata['ajax_url'], podcast_display_id, data=urlencode_postdata({
                'action': 'checkpodcast',
                'cookie': endpoint_metadata['cookie'],
                'post_id': episode_id,
                'podcast_id': podcast_id,
            }))
```

File: yt_dlp/extractor/ilpost.py [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+        if not episode:
+            raise ExtractorError('Episode could not be extracted')
+        return {
+            'id': episode_id,
+            'display_id': podcast_display_id,
+            'series_id': podcast_id,
+            'vcodec': 'none',
+            **traverse_obj(episode, {
+                'title': 'title',
+                'description': 'description',
```
Comment (seproDev) [inline yt_dlp/extractor/ilpost.py:58]:
```suggestion
                'title': ('title', {str}),
                'description': ('description', {str}),
```

**Review by seproDev — APPROVED** [review]

**Comment by seproDev** [comment]
Also, for the future, please don't force push.

File: yt_dlp/extractor/ilpost.py [hunk 6]
```diff
@@ -0,0 +1,61 @@
+import functools
+
+from .common import InfoExtractor
+from ..utils import ExtractorError, float_or_none, int_or_none, url_or_none, urlencode_postdata
+from ..utils.traversal import traverse_obj
```
Comment (bashonly) [inline yt_dlp/extractor/ilpost.py:5]:
```suggestion
import functools

from .common import InfoExtractor
from ..utils import (
    ExtractorError,
    float_or_none,
    int_or_none,
    url_or_none,
    urlencode_postdata,
)
from ..utils.traversal import traverse_obj
```

File: yt_dlp/extractor/ilpost.py [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+
+
+class IlPostExtractorIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?ilpost\.it/episodes/(?P<id>[^/?]+).*'
```
Comment (bashonly) [inline yt_dlp/extractor/ilpost.py:9]:
```suggestion
    _VALID_URL = r'https?://(?:www\.)?ilpost\.it/episodes/(?P<id>[^/?#]+)'
```

File: yt_dlp/extractor/ilpost.py [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+            'upload_date': '20231229',
+            'duration': 2495.0,
+            'availability': 'public',
+            'series_id': '235598',
+            'description': '',
+        }
+    }]
+
+    def _real_extract(self, url):
+        podcast_display_id = self._match_id(url)
```
Comment (bashonly) [inline yt_dlp/extractor/ilpost.py:29]: `display_id` is the convention, shorter, and more distinct from `podcast_id`
```suggestion
        display_id = self._match_id(url)
```

File: yt_dlp/extractor/ilpost.py [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+                'post_id': episode_id,
+                'podcast_id': podcast_id,
+            }))
+        episode = traverse_obj(podcast_metadata, (
+            'data', 'postcastList', lambda _, v: str(v['id']) == episode_id, {dict}), get_all=False)
+        if not episode:
+            raise ExtractorError('Episode could not be extracted')
+        return {
+            'id': episode_id,
+            'display_id': podcast_display_id,
```
Comment (bashonly) [inline yt_dlp/extractor/ilpost.py:49]:
```suggestion
            'display_id': display_id,
```

File: yt_dlp/extractor/ilpost.py [hunk 10]
```diff
@@ -0,0 +1,61 @@
...
+    def _real_extract(self, url):
+        podcast_display_id = self._match_id(url)
+        webpage = self._download_webpage(url, podcast_display_id)
+
+        endpoint_metadata = self._search_json(
+            r'var\s+ilpostpodcast\s*=', webpage, 'metadata', podcast_display_id)
+        episode_id = endpoint_metadata['post_id']
+        podcast_id = endpoint_metadata['podcast_id']
+        podcast_metadata = self._download_json(
+            endpoint_metadata['ajax_url'], podcast_display_id, data=urlencode_postdata({
```
Comment (bashonly) [inline yt_dlp/extractor/ilpost.py:37]:
```suggestion
            endpoint_metadata['ajax_url'], display_id, data=urlencode_postdata({
```

File: yt_dlp/extractor/ilpost.py [hunk 11]
```diff
@@ -0,0 +1,61 @@
...
+            'description': '',
+        }
+    }]
+
+    def _real_extract(self, url):
+        podcast_display_id = self._match_id(url)
+        webpage = self._download_webpage(url, podcast_display_id)
+
+        endpoint_metadata = self._search_json(
+            r'var\s+ilpostpodcast\s*=', webpage, 'metadata', podcast_display_id)
```
Comment (bashonly) [inline yt_dlp/extractor/ilpost.py:33]:
```suggestion
            r'var\s+ilpostpodcast\s*=', webpage, 'metadata', display_id)
```

File: yt_dlp/extractor/_extractors.py [hunk 12]
```diff
@@ -787,6 +787,7 @@
     IHeartRadioIE,
     IHeartRadioPodcastIE,
 )
+from .ilpost import IlPostExtractorIE
```
Comment (bashonly) [inline yt_dlp/extractor/_extractors.py:790]: `Extractor` is redundant
```suggestion
from .ilpost import IlPostIE
```

File: yt_dlp/extractor/ilpost.py [hunk 13]
```diff
@@ -0,0 +1,61 @@
+import functools
+
+from .common import InfoExtractor
+from ..utils import ExtractorError, float_or_none, int_or_none, url_or_none, urlencode_postdata
+from ..utils.traversal import traverse_obj
+
+
+class IlPostExtractorIE(InfoExtractor):
```
Comment (bashonly) [inline yt_dlp/extractor/ilpost.py:8]:
```suggestion
class IlPostIE(InfoExtractor):
```

**Review by bashonly — CHANGES_REQUESTED** [review]

File: yt_dlp/extractor/ilpost.py [hunk 14]
```diff
@@ -0,0 +1,61 @@
...
+            'duration': 2495.0,
+            'availability': 'public',
+            'series_id': '235598',
+            'description': '',
+        }
+    }]
+
+    def _real_extract(self, url):
+        podcast_display_id = self._match_id(url)
+        webpage = self._download_webpage(url, podcast_display_id)
```
Comment (bashonly) [inline yt_dlp/extractor/ilpost.py:30]:
```suggestion
        webpage = self._download_webpage(url, display_id)
```

**Review by bashonly — APPROVED** [review]

**Pull request merged**